src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  width: $daysPerRow * $dayShift;
+
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+  gap: #{$dayGap};
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background: #eee;
+    padding: $dayPadding;
+    @include square (100px, 100px);
+
+    border: 1px solid #000;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day}):before {
+        content: "#{$day}";
+      }
+    }
+  }
+
+  @for $last-day from 28 through 31 {
+    &--month-length--#{$last-day} {
+      .calendar__day:nth-child(n + #{$last-day + 1 }) {
+        display: none;
+      }
+    }
+  }
+
+  @each $wkDay, $dayNum in $week-days {
+    &--start-day--#{$wkDay} {
+      .calendar__day:first-child {
+        margin-left: ($dayShift * $dayNum);
+      }
+    }
+  }
+
+  &__day:hover {
+    background-color: pink;
+    cursor: pointer;
+    transform: translateY(-20px);
+    transition: all 0.5s;
+  }
```

Comment:
you're repeating the code
advise you to nest this style under `&__day` above, then it would be more clear where it belongs
also suggest you to use transition not on the :hover state, but on the class - then both hover and un-hover will be smooth.
```suggestion
```